src/init/starship.ps1

```diff
@@ -14,9 +16,38 @@ function global:prompt {
     $current_directory = (Convert-Path -LiteralPath $PWD)
     
     # If an external command has not been executed, then the $LASTEXITCODE will be $null.
-    # For the purposes of the prompt, replace this with a 0 exit code so that the prompt
-    # doesn't show the last command as a failure (because it had a non-zero exit code).
-    $lastExitCodeForPrompt = if ($origLastExitCode) { $origLastExitCode } else { 0 }
+    # it isn't populated or modified until you execute a Powershell task that returns an exit
+    # code e.g. On a fresh Powershell console:
+    # > $LASTEXITCODE -eq $null
+    # True
+    # In the same console after running a Powershell cmdlet:
+    # > gci ~/non_existing_folder
+    # ...
+    # > $LASTEXITCODE -eq $null
+    # True
+    # It only gets populated when running an external executable that returns an exit code, e. g.
+    # > bash -c 'exit 5'
+    # > $LASTEXITCODE -eq $null
+    # False
+    # > $LASTEXITCODE
+    # 5
+    # Whe start from the premise that the command executed correctly, which covers also the fresh console.
+    $lastExitCodeForPrompt = 0
+
+    # In case we have a False on the Dollar hook, we know there's an error.
+    if( -not $origDollarQuestion){
```

Comment:
Could this reuse ` if ($lastCmd = Get-History -Count 1) {` from below?

Reply:
Moved the logic to andle $LASTEXITCODE inside the $lastCmd if block as suggested to take advantage of that variable.